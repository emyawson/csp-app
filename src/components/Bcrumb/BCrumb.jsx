// Modules

import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';

import { withRouter } from 'react-router-dom';
// eslint-disable-next-line no-unused-vars
import Breadcrumb from '@ingka/breadcrumb';
import Hyperlink from '@ingka/hyperlink';
import './BCrumb.scss';

// @ingka
import '@ingka/hyperlink/style.scss';
import '@ingka/focus/style.scss';
import '@ingka/breadcrumb/style.scss';
import { ROUTE_BASE, ROUTE_CUSTOMER_SERVICE, ROUTE_NL } from '../../constants/routes';
import { phrase } from '../../utils/phrase';
import { selectPages } from '../../store/selectors';

const BCrumb = () => {
  const pages = useSelector(selectPages);
  let lastCrumb;
  pages.error ? lastCrumb = pages.error.message : lastCrumb = pages.headline;

  const bcrumbPathName = [
    { url: ROUTE_NL, location: phrase.rootHome },
    { url: ROUTE_CUSTOMER_SERVICE, location: phrase.rootCustomerService },
    { url: ROUTE_BASE, location: lastCrumb },
  ];

  return (
    <Breadcrumb className="breadcrumb" data-cytest="breadcrumb">
      {bcrumbPathName.map((item, index) => (
        <Hyperlink url={item.url} key={index}>
          <span className="breadcrumb__item" data-cytes='crumb'>{item.location}</span>
        </Hyperlink>
      ))}
    </Breadcrumb>
  );
};

BCrumb.propTypes = {
  location: PropTypes.object.isRequired,
};

export default withRouter(BCrumb);
