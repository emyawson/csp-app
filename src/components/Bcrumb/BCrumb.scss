@import '../../index';

.breadcrumb__item {
  color: $colour-neutral-grey-900;
  font-size: $font-XS;
  line-height: 18px;
}

