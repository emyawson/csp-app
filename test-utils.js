import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { Router } from 'react-router-dom';
import { connectRouter } from 'connected-react-router';
import { render } from '@testing-library/react';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import { history } from './src/store';
import reducers from './src/store/reducer/index';

const createRootReducer = (reducerHistory) => combineReducers({
  router: connectRouter(reducerHistory),
  ...reducers,
});

// Setup redux persist
const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
};
const persistedReducer = persistReducer(
  persistConfig,
  createRootReducer(history),
);

const renderWithRouter = (ui, options = {}, initialState) => ({
  ...render(
      <Provider store={createStore(persistedReducer, initialState)}>
        <Router history={history}>{ui}</Router>
      </Provider>,
  ),
});

const AllTheProviders = () => {};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, renderWithRouter };
