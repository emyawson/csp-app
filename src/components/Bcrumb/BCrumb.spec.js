import React from 'react';
// import { render, screen } from "@testing-library/react";
import BCrumb from './BCrumb.jsx';
// import { renderWithRouter } from "../test-utils";
import { renderWithRouter } from '../../../test-utils';

const defaultState = {};

const renderComponent = (state = {}) => {
  const combinedState = {
    ...defaultState,
    ...state,
  };
  return renderWithRouter(<BCrumb />, combinedState);
};

test('renders <BCrumb/>', () => {
  const { container } = renderComponent();
  expect(container).toMatchSnapshot();

  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});
