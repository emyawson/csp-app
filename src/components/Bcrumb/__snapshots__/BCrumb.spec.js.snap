// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <BCrumb/> 1`] = `
<div>
  <div
    class="breadcrumb"
  >
    <nav
      aria-label="Breadcrumb"
      class="breadcrumb__nav"
      data-cytest="breadcrumb"
      role="navigation"
    >
      <ol
        class="breadcrumb__list"
        itemprop="breadcrumb"
        itemscope=""
        itemtype="http://schema.org/BreadcrumbList"
      >
        <li
          class="breadcrumb__list-item"
          itemprop="itemListElement"
          itemscope=""
          itemtype="http://schema.org/ListItem"
        >
           
          <a
            class="breadcrumb__link link link--black"
            href="/nl/nl"
            itemprop="item"
          >
            
            <span
              class="breadcrumb__item"
              data-cytes="crumb"
            >
              IKEA
            </span>
          </a>
           
          <meta
            content="1"
            itemprop="position"
          />
        </li>
        <li
          class="breadcrumb__list-item"
          itemprop="itemListElement"
          itemscope=""
          itemtype="http://schema.org/ListItem"
        >
           
          <a
            class="breadcrumb__link link link--black"
            href="/nl/nl/customer-service"
            itemprop="item"
          >
            
            <span
              class="breadcrumb__item"
              data-cytes="crumb"
            >
              Klantenservice
            </span>
          </a>
           
          <meta
            content="2"
            itemprop="position"
          />
        </li>
        <li
          class="breadcrumb__list-item"
          itemprop="itemListElement"
          itemscope=""
          itemtype="http://schema.org/ListItem"
        >
           
          <a
            class="breadcrumb__link link link--black"
            href="/nl/nl/customer-service/support"
            itemprop="item"
          >
            
            <span
              class="breadcrumb__item"
              data-cytes="crumb"
            />
          </a>
           
          <meta
            content="3"
            itemprop="position"
          />
        </li>
      </ol>
    </nav>
  </div>
</div>
`;
